vpr/src/route/clock_connection_builders.cpp

```diff
@@ -76,13 +74,13 @@ void RoutingToClockConnection::create_switches(const ClockRRGraphBuilder& clock_
         // Connect to x-channel wires
         unsigned num_wires_x = x_wire_indices.size() * fc;
         for (size_t i = 0; i < num_wires_x; i++) {
-            clock_graph.add_edge(rr_edges_to_create, x_wire_indices[i], clock_index, arch_switch_idx);
+            clock_graph.add_edge(rr_edges_to_create, size_t(x_wire_indices[i]), clock_index, arch_switch_idx);
```

Comment:
Do we need the size_t cast?  Can we change the clock_graph.add_edge function to take an RRNodeId and get rid of the cast?

Reply:
@vaughnbetz 

Strongly Agree. I did not do it because of the ``clock_graph.add_edge()`` function relies on the data structure ``t_rr_edge_info``

https://github.com/verilog-to-routing/vtr-verilog-to-routing/blob/c06800fb49d2732a0fcedbb2f542613fedd6a330/vpr/src/route/rr_edge.h#L4-L14

where you can see that the data structure uses the integer as node id.
The ``t_rr_edge_info`` has been used in many functions across ``rr_graph2.cpp``, ``rr_graph.cpp`` as well the clock builder functions, whenever an edge is added.
You can see it will require massive changes to the code base.

To avoid big changes in this PR, I added the TODO comment to the ``t_rr_edge_info``, as a reminder that we must change it before finishing the refactoring.
This should be the topic of my next PR after my current two PRs are merged.
See if this is o.k. for you.

Reply:
OK, thanks.